android/src/com/mapswithme/maps/editor/ViralFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.mapswithme.maps.editor;
+
+import android.annotation.SuppressLint;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import java.util.Random;
+
+import com.mapswithme.maps.MwmApplication;
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.base.BaseMwmDialogFragment;
+import com.mapswithme.util.ConnectionState;
+
+public class ViralFragment extends BaseMwmDialogFragment
+{
+  private static final String EXTRA_CONTRATS_SHOWN = "CongratsShown";
+  private TextView mViral;
+
+  @StringRes
+  private static int sViralText;
+
+  public static boolean shouldDisplay()
+  {
+    return !MwmApplication.prefs().contains(EXTRA_CONTRATS_SHOWN) &&
+           Editor.nativeGetStats()[0] >= 2 &&
+           ConnectionState.isConnected();
+  }
+
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(Bundle savedInstanceState)
+  {
+    MwmApplication.prefs().edit().putBoolean(EXTRA_CONTRATS_SHOWN, true).apply();
```

Comment:
Конфиг же про нативные сеттинги, в префсы по месту использования много где пишется.
> On Mar 18, 2016, at 5:42 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/editor/ViralFragment.java <https://github.com/mapsme/omim/pull/2338#discussion_r56665955>:
> 
> > +
> > +  @StringRes
> > +  private static int sViralText;
> > +
> > +  public static boolean shouldDisplay()
> > +  {
> > +    return !MwmApplication.prefs().contains(EXTRA_CONTRATS_SHOWN) &&
> > +           Editor.nativeGetStats()[0] >= 2 &&
> > +           ConnectionState.isConnected();
> > +  }
> > +
> > +  @NonNull
> > +  @Override
> > +  public Dialog onCreateDialog(Bundle savedInstanceState)
> > +  {
> > +    MwmApplication.prefs().edit().putBoolean(EXTRA_CONTRATS_SHOWN, true).apply();
> Такие флажки в Config лежат. Может, тоже туда вынести?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2338/files/49a98513afd524db41d893d7cec128cfb271218e..b0dcfabe82c6bad90154275d1b20f2651bbdf1b1#r56665955>